src/ion/moves.rs

```diff
@@ -1001,6 +830,194 @@ impl<'a, F: Function> Env<'a, F> {
             }
         }
 
-        edits
+        self.ctx.output.edits.clear();
+        self.ctx.output.edits.extend(edits.drain_edits());
+
+        self.ctx.scratch_moves.inserted_moves = inserted_moves;
+        self.ctx.scratch_moves.edits = edits;
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct MoveCtx {
+    inserted_moves: InsertedMoves,
+    edits: Edits,
+    inter_block_sources: Vec<(VRegIndex, Allocation)>,
+    block_param_sources: FxHashMap<BlockparamSourceKey, Allocation>,
+    inter_block_dests: Vec<InterBlockDest>,
+    block_param_dests: Vec<BlockparamDest>,
+    reuse_input_insts: Vec<Inst>,
+}
+
+/// Buffered information about the previous liverange that was processed.
+struct PrevBuffer {
```

Comment:
Good idea moving this out of the function-local context, but can we have it in a separate PR (with code motion but no other changes) to make review easier / make any diffs apparent?